src/Analyser/MutatingScope.php

```diff
@@ -1051,12 +1051,24 @@ private function resolveType(Expr $node): Type
 			} else {
 				$right = $node->right;
 			}
+			$rightType = $this->getType($right);
+
+			$stringType = $rightType->toString();
+			if (
+				$node instanceof Node\Expr\BinaryOp\Mod
+				|| $node instanceof Node\Expr\AssignOp\Mod
+			) {
+				if ($stringType instanceof ConstantScalarType && $stringType->getValue() === '1') {
```

Comment:
This logic is only going to be executed for ConstantStringType. But the same logic works for `1.0` as well: https://3v4l.org/CYeUe

I'd rather be if you called `->toInteger()` and checked ConstantIntegerType afterwards.